loki/transformations/build_system/tests/test_dependency.py

```diff
@@ -415,40 +419,80 @@ def test_dependency_transformation_module_wrap(frontend, use_scheduler, tmp_path
 END SUBROUTINE kernel
     """.strip()
 
+    other_kernel_fcode = """
+SUBROUTINE other_kernel(a, b, c)
+  INTEGER, INTENT(INOUT) :: a, b, c
+
+  a = 1
+  b = 2
+  c = 3
+END SUBROUTINE other_kernel
+    """.strip()
+
     transformations = (
-        ModuleWrapTransformation(module_suffix='_mod'),
-        DependencyTransformation(suffix='_test', module_suffix='_mod')
+        ModuleWrapTransformation(module_suffix='_mod', replace_ignore_items=replace_ignore_items),
+        DependencyTransformation(suffix='_test', module_suffix='_mod', replace_ignore_items=replace_ignore_items)
     )
 
     if use_scheduler:
         (tmp_path/'kernel.F90').write_text(kernel_fcode)
+        (tmp_path/'other_kernel.F90').write_text(other_kernel_fcode)
         (tmp_path/'driver.F90').write_text(driver_fcode)
+
+        _config = copy.deepcopy(config)
+        if not replace_ignore_items:
+            _config['default'].update({'block': ['kernel']})
+
         scheduler = Scheduler(
-            paths=[tmp_path], config=SchedulerConfig.from_dict(config), frontend=frontend, xmods=[tmp_path]
+            paths=[tmp_path], config=SchedulerConfig.from_dict(_config), frontend=frontend, xmods=[tmp_path]
         )
         for transformation in transformations:
             scheduler.process(transformation)
 
-        kernel = scheduler['kernel_test_mod#kernel_test'].source
+        if replace_ignore_items:
+            kernel = scheduler['kernel_test_mod#kernel_test'].source
+        else:
+            with pytest.raises(AttributeError):
+                kernel = scheduler['#kernel'].source
+                kernel = scheduler['kernel_mod#kernel'].source
+                kernel = scheduler['kernel_test_mod#kernel_test'].source
```

Comment:
Don't you need to separate these to test that the exception is raised for each of them? I expect this to bail out on the first exception occurence
```suggestion
            for item_name in ['#kernel', 'kernel_mod#kernel', 'kernel_test_mod#kernel_test']:
                with pytest.raises(AttributeError):
                    _ = scheduler[item_name].source
```

Reply:
Yes good catch! I'll update the test.